### PR TITLE
Add user response option to Turn Annotations Static Task

### PR DESCRIPTION
**Pull request by meganung**
This PR adds the option to have a response field textbox for each utterance so we can collect the user’s response (chat history is not updated with the user’s response).
- Validation of the user’s text input: requires more than 10 characters, at least 1 vowel, and at least 2 words

Annotation buckets can also be optional, if no annotation buckets file path is passed in there will be no annotation buckets.

Screen Shot 2021-01-04 at 2 35 09 PM

### CODE REVIEW

**Comment by EricMichaelSmith** [comment]
(adding @JackUrb as well in case he has thoughts on the Javascript)

File: parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx
Comment (JackUrb) [inline parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:224]: Nit: I'm not sure this is quite saying the right thing to a worker due to the chaining ands, though I'm not really sure what asking them to "Please do not ... respond to each message as you would!" is even implying. Perhaps "Please do not copy and paste. You must manually respond to each message." is better?

File: parlai/crowdsourcing/tasks/turn_annotations_static/turn_annotations_blueprint.py
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/turn_annotations_static/turn_annotations_blueprint.py:82]: I think this is a good change - it's good to have no checkboxes by default. In fact, I just made that change to the `model_chat` task as well :)

File: parlai/crowdsourcing/tasks/turn_annotations_static/turn_annotations_blueprint.py
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/turn_annotations_static/turn_annotations_blueprint.py:89]: nit: it might be good to be more specific by saying that this will add a text box, for use by the crowdsource worker in responding to the message

File: parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:64]: nit: check the spelling of "Response"

**Review by EricMichaelSmith — APPROVED** [review]
Great, nice to have! Approving with a few small notes. But before merging in, can you try merging from master (`git merge master`)? I'm surprised that only 5 CI checks are running - usually roughly ~20 run, and it'd be good to make sure that the crowdsourcing CI check still passes with these changes, for example

File: parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx
Comment (jxmsML) [reply parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:39]: is it possible to pass those checkers from backend to front-end? @JackUrb

Comment (jxmsML) [reply parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:39]: minor: it would be better to set thresholds of charCount, wordCount, and numVowels configurable.

Comment (EricMichaelSmith) [reply parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:39]: Also, you might want to talk to @mojtaba-komeili - I know he's also doing per-message checking on the frontend, so it'd be good to reuse code with him if possible

Comment (JackUrb) [reply parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx]: In this case, as it's a static task there's no real way to communicate with the backend. The validator needs to be written in JS, but it could be configurable.

**Pull request closed without merging**
